### PR TITLE
Code cleanup - mostly regarding deprecated api's.

### PR DESCRIPTION
**Pull request by pprucia**
Code cleanup - mostly regarding deprecated api's. Devices with newer api available will use it, older ones will use legacy api.
#### APK
[Download build #8800](http://192.168.10.18:8080/job/Pull%20Request%20Builder/8800/artifact/build/artifact/wire-dev-PR810-8800.apk)

### CODE REVIEW

**Comment by AndroidBob** [comment]
Can one of the admins verify this patch?

**Comment by hakonbo** [comment]
build again

File: app/src/main/java/com/waz/zclient/pages/main/conversationlist/views/row/ConversationListRow.java [hunk 1]
```diff
@@ -130,7 +131,12 @@ public ConversationListRow(Context context) {
         super(context);
         this.isSwipeable = true;
 
-        normalTextColor = getResources().getColor(R.color.list_font_color);
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) {
+            //noinspection deprecation
+            normalTextColor = getResources().getColor(R.color.list_font_color);
+        } else {
+            normalTextColor = getResources().getColor(R.color.list_font_color,getContext().getTheme());
```
Comment (hakonbo) [inline app/src/main/java/com/waz/zclient/pages/main/conversationlist/views/row/ConversationListRow.java]: Can you add a space after comma in the argument list. Our build server rejects the PRs unless checkstyle/lint/pmd passes. You can check yourself by running `./gradlew ci`.

Comment (pprucia) [inline app/src/main/java/com/waz/zclient/pages/main/conversationlist/views/row/ConversationListRow.java]: Done.

**Comment by hakonbo** [comment]
build again

**Comment by hakonbo** [comment]
Thanks @pprucia

**Comment by wire-android** [comment]
it would be nicer to use helper `getColor` function encapsulating version check instead of adding those `ifs` all over the place

**Comment by pprucia** [comment]
Agree, I saw that in scala code, plan to do it in java in some time.

**Pull request merged**